src/__test__/userController.test.ts

```diff
@@ -234,6 +230,48 @@ describe('User Login Tests', () => {
   });
 
 
+  it('should log in a buyer with valid credentials', async () => {
+    const formData = {
+      name: 'Buyer',
+      permissions: ['test-permission1', 'test-permission2'],
+    };
+  
+    // Create the role first
+    const roleResponse = await request(app).post('/api/v1/roles/create_role').send(formData);
+    
+    const userData = {
+      firstName: 'Test',
+      lastName: 'User',
+      email: 'test2@gmail.com',
+      password: 'TestPassword123',
+      userType: roleResponse.body.id
+    };
+    await request(app).post('/api/v1/register').send(userData);
+  
+    const updatedUser = await userRepository.findOne({ where: { email: userData.email } });
+    if (updatedUser) {
+      updatedUser.isVerified = true;
+      await userRepository.save(updatedUser);
+      
+      const loginResponse = await request(app).post('/api/v1/login').send({
+        email: userData.email,
+        password: userData.password,
+      });
+      
+      expect(loginResponse.status).toBe(200);
+      expect(loginResponse.body.token).toBeDefined();
+      expect(loginResponse.body.message).toBe('Buyer Logged in successfully');
+  
+      // Decode the token and check its properties
+      const decodedToken = jwt.decode(loginResponse.body.token);
+      expect(decodedToken).toHaveProperty('userId');
+      expect(decodedToken).toHaveProperty('iat');
+      expect(decodedToken).toHaveProperty('exp');
+    } else {
+      throw new Error('User not found');
+    }
```

Comment:
We don't need to throw an error in test, expect what to be expected otherwise this else is kinda useless